### DISCUSSION TITLE
Is changing textures gradually possible?

### DISCUSSION THREAD

**Discussion by danilokrulj** [google/model-viewer#4827] [Q&A] [answered]
Title describes the question really. Is changing textures gradually, instead of instantly, possible? And if it is, how could it be done?

**Comment by elalish** [comment, marked as answer]
No, we don't have an API for that, mostly because it's actually pretty hard. Textures can be different sizes, so you'd need a shader to mix them together and render to a third texture every frame. I think it would introduce too many edge cases. BaseColorFactor is multiplied onto the BaseColorTexture, so you could fade to black, switch, then fade back in.

**Reply by danilokrulj** [reply]
Thank you for your reply. I taught it was hard to do and I was also suspecting that it would take up a lot of resources but I still had to ask. Hope never dies :)
